src/main/java/org/apache/sling/auth/oauth_client/impl/OidcAuthenticationHandler.java

```diff
@@ -187,32 +198,40 @@ public AuthenticationInfo extractCredentials(@NotNull HttpServletRequest request
             return authInfo;
         }
 
-        //The request is not authenticated. 
-        // 1. Check if the State cookie match with the state in the request received from the idp
+        //The request is not authenticated.
+        // 1. Extract nonce cookie and state cookie from the request
         StringBuffer requestURL = request.getRequestURL();
         if ( request.getQueryString() != null )
             requestURL.append('?').append(request.getQueryString());
 
         Optional<OAuthState> clientState; //state returned by the idp in the redirect request
         String authCode; //authorization code returned by the idp in the redirect request
         Cookie stateCookie;
+        Cookie nonceCookie;
```

Comment:
I would prefer to use a single cookie, if at all possible. This makes the response payload smaller.

Reply:
@rombert @anchela 
Currently in oauth flow we send several information in the state:
- perRequestKey (random value to be verified)
- connectionName
- redirect
The perRequestKey is also sent in one encrypted cookie to be verified for CSRF protection.
During OIDC authentication flow, additional information (nonce and code verifier) are sent into separated cookies.

If you agree we may send a single json encrypted cookie with all the information required by both the flows, and send only the perRequestKey in the state.

Reply:
In [rfc6265](https://datatracker.ietf.org/doc/html/rfc6265?#section-6.1) read that these are the limits that browsers SHOULD support:
```
   o  At least 4096 bytes per cookie (as measured by the sum of the
      length of the cookie's name, value, and attributes).
   o  At least 50 cookies per domain.
   o  At least 3000 cookies total.
```
If we want to be in a safer side and limit the number of cookies as well we may:
- keep the current setup for OAuth
- Modify the current code to have only one new cookie for nonce and code verifier

Reply:
If we are sure we can fit within 4096 byes I would suggest that we keep one cookie, this should make the payload smaller and the code easier to understand.

I am not sure if it makes sense to have different coookie names for the OAuth servlets vs the Authentication Handler, to simplify the code. Just an idea, not a suggestion :-)